tests/disas/pcc-insertlane-x64.wat

```diff
@@ -143,8 +143,8 @@
 ;;       movq    %rsp, %rbp
 ;;       movl    %edx, %r10d
 ;;       movq    0x40(%rdi), %r11
-;;       movdqu  (%r11, %r10), %xmm6
```

Comment:
The old version seems wrong... this _should_ be a `movsd` I believe.

Reply:
Yeah definitely wrong, `XmmMem` auto-converted to `Xmm` always loads 128-bits, and this is just an incorrect translation and means that on-the-boundary-of-linear-memory loads of f32/f64 will be incorrectly identified as out-of-bounds.

Reply:
For this, mind adding [this test](https://github.com/alexcrichton/wasmtime/commit/3a34e2bac097ddce38da0fa0b3f3ab7886696c4a) to this PR?

Reply:
Added. It doesn't fail, though?

Reply:
(Maybe it would have failed with the old version?)